### PR TITLE
fix(calcite-input): number inputs with step="any" should allow decimals

### PR DESCRIPTION
**Pull request by eriklharper**
**Related Issue:** #2764

This pr allows number inputs with `step="any"` to input decimal values akin to the native number input.

### CODE REVIEW

**Comment by eriklharper** [comment]
test is forthcoming, stay tuned!

File: src/components/calcite-input/calcite-input.tsx [hunk 1]
```diff
@@ -484,7 +483,7 @@ export class CalciteInput {
     const decimalSeparator = getDecimalSeparator(this.locale);
     if (
       event.key === decimalSeparator &&
-      isValidDecimal(this.step === "any" ? 1 : (this.step as number))
+      (this.step === "any" ? true : isValidDecimal(this.step))
```
Comment (jcfranco) [inline src/components/calcite-input/calcite-input.tsx:486]: Nitpick: This can be simplified to `this.step === "any" || isValidDecimal(this.step)`

**Review by jcfranco — APPROVED** [review]
✨🥇 ✨

File: src/components/calcite-input/calcite-input.tsx (hunk 1 above)
Comment (eriklharper) [reply src/components/calcite-input/calcite-input.tsx:486]: Yeah for some reason I had it that way and it was allowing decimals in integer-only mode, but now that I add it back in I'm seeing it work.  Must have been a dev server problem where the changes weren't loading right or something.

**Pull request merged**